packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1387,6 +1426,9 @@ abstract class FlutterCommand extends Command<void> {
   @mustCallSuper
   Future<FlutterCommandResult> verifyThenRunCommand(String? commandPath) async {
     globals.preRunValidator.validate();
+
+    startFindingWirelessDevices();
```

Comment:
what if, instead of always calling this method and requiring methods that don't need it to override it, we add a boolean field to `FlutterCommand`: `final bool usesWirelessDevices = false;`

Then commands that require it must explicitly opt in. This way someone adding a new command does not have to consciously opt out.